components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -1467,6 +1470,105 @@ private static void prepareForUpdateSwagger(String apiId, APIDefinitionValidatio
+                JSONObject existingEndpointConfigJson = (JSONObject) new JSONParser()
+                        .parse(existingEndpointConfigString);
+                if (newProductionEndpointJson != null) {
+                    if (existingEndpointConfigJson.get(APIConstants.ENDPOINT_PRODUCTION_ENDPOINTS) != null) {
+                        //put as a value under the ENDPOINT_PRODUCTION_ENDPOINTS key
+                        //if loadbalance endpoints, get relevant jsonobject from array
+                        if (existingEndpointConfigJson.get(APIConstants.API_ENDPOINT_CONFIG_PROTOCOL_TYPE)
+                                .equals(APIConstants.ENDPOINT_TYPE_LOADBALANCE)) {
+                            JSONArray productionConfigsJson = (JSONArray) existingEndpointConfigJson
+                                    .get(APIConstants.ENDPOINT_PRODUCTION_ENDPOINTS);
+                            for (int i = 0; i < productionConfigsJson.size(); i++) {
+                                if (!(((JSONObject) productionConfigsJson.get(i)).containsKey(APIConstants
+                                        .API_ENDPOINT_CONFIG_PROTOCOL_TYPE))) {
+                                    if (newProductionEndpointJson.has(APIConstants
+                                            .ADVANCE_ENDPOINT_CONFIG)) {
+                                        JsonNode advanceConfig = newProductionEndpointJson
+                                                .get(APIConstants.ADVANCE_ENDPOINT_CONFIG);
+                                        ((JSONObject) productionConfigsJson.get(i))
+                                                .put(APIConstants.ADVANCE_ENDPOINT_CONFIG, advanceConfig);
+                                    } else {
+                                        ((JSONObject) productionConfigsJson.get(i))
+                                                .remove(APIConstants.ADVANCE_ENDPOINT_CONFIG);
+                                    }
+                                    break;
+                                }
+                            }
+                            existingEndpointConfigJson.put(APIConstants.ENDPOINT_PRODUCTION_ENDPOINTS,
+                                    productionConfigsJson);
+                        } else {
+                            JSONObject productionConfigsJson = (JSONObject) existingEndpointConfigJson
+                                    .get(APIConstants.ENDPOINT_PRODUCTION_ENDPOINTS);
+                            if (newProductionEndpointJson.has(APIConstants.ADVANCE_ENDPOINT_CONFIG)) {
+                                JsonNode advanceConfig = newProductionEndpointJson
+                                        .get(APIConstants.ADVANCE_ENDPOINT_CONFIG);
+                                productionConfigsJson.put(APIConstants.ADVANCE_ENDPOINT_CONFIG, advanceConfig);
+                            } else {
+                                productionConfigsJson.remove(APIConstants.ADVANCE_ENDPOINT_CONFIG);
```

Comment:
What happens to the advanced endpoint configs (synapse) added through the UI in this case? Will they also get removed if there's no advanced endpoint config (CC) included in the swagger?

Reply:
Those details are not added under the advanceEndpointConfigs in the swagger definition